src/util/pool_timer.cpp

```diff
@@ -0,0 +1,148 @@
+//  Copyright (c) 2016 Bibek Wagle
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_fwd.hpp>
+#include <hpx/exception.hpp>
```

Comment:
We're trying to reduce the use of hpx_fwd.hpp, replacing it by the single headers declaring the things really needed.